x/muxer/muxer.go

```diff
@@ -394,19 +395,16 @@ func (m *muxer) GetPluginInfo(ctx context.Context, e *emptypb.Empty) (*rpc.Plugi
 }
 
 func (m *muxer) Attach(ctx context.Context, req *rpc.PluginAttach) (*emptypb.Empty, error) {
-	host, err := provider.NewHostClient(req.GetAddress())
-	if err != nil {
-		return nil, err
-	}
-	if m.host != nil {
-		if err := m.host.Close(); err != nil {
-			return nil, err
+	attach := make([]func() error, len(m.servers))
+	for i, s := range m.servers {
+		s := s
+		attach[i] = func() error {
+			_, err := s.Attach(ctx, req)
+			return err
 		}
 	}
-	// Here we override the underlying host. This should replace the host instance of
-	// each subsidiary provider.
-	*m.host = *host
```

Comment:
does it matter that we're no longer setting `m.host`? Do we still need that field?

Reply:
Good catch. I believe that it does matter at the margin. Fixed in the next commit.